src/routes/forgot-password/components/ForgotPassword.js

```diff
@@ -0,0 +1,106 @@
+const React = require('react');
+const T = require('prop-types');
+const StrangeForms = require('strange-forms');
+const IsEmail = require('utils/is-email');
+
+module.exports = class ForgotPassword extends StrangeForms(React.Component) {
+
+    static propTypes = {
+        requestReset: T.func.isRequired,
+        errorMessage: T.string
+    };
+
+    constructor(props) {
+
+        super(props);
+
+        this.state = {
+            email: '',
+            isBlurred: {
+                email: false
+            }
+        };
+
+        this.strangeForm({
+            fields: ['email'],
+            get: (someProps, field) => this.state[field],
+            act: this.formField.bind(this),
+            getFormValue: this.getFormValue.bind(this)
+        });
+    }
+
+    getFormValue(e) {
+
+        return e.target.value || '';
+    }
+
+    formField(field, value) {
+
+        this.setState({ [field]: value });
+    }
+
+    fieldBlurred = (ev) => {
+
+        const isBlurred = { ...this.state.isBlurred };
+        const field = ev.target.id;
+
+        isBlurred[field] = true;
+
+        this.setState({ isBlurred });
+    }
+
+    invalidEmail = () => {
```

Comment:
For functions that return a boolean, I like tacking on `is` to the beginning to make the function sound like a question.  Helps with readability.